electron/app/services/cloud/login.ts

```diff
@@ -0,0 +1,28 @@
+import {shell} from 'electron';
+
+import {CloudLoginResponse} from '@shared/models/cloud';
+
+import {getAuthenticator} from './authenticator';
+import {serializeUser} from './user';
+
+export const cloudLogin = async (): Promise<CloudLoginResponse> => {
+  const authenticator = await getAuthenticator();
+  if (!authenticator) {
+    throw new Error('Something went wrong with the authenticator');
+  }
+  const loginResponse = await authenticator.login('device code');
+  if (!loginResponse.handle) {
+    throw new Error('Something went wrong with the login response');
+  }
+  shell.openExternal(loginResponse.handle.verification_uri_complete);
+  const user = await loginResponse.onDone;
+  if (!user) {
+    throw new Error('Login to Cloud has failed');
+  }
```

Comment:
can we get any more context/details for any of the above error conditions and write those to the logs!?